pkg/deploy/controller/lifecycle/manager.go

```diff
@@ -0,0 +1,164 @@
+package lifecycle
+
+import (
+	"fmt"
+
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+
+	deployapi "github.com/openshift/origin/pkg/deploy/api"
+)
+
+// Status represents the status of a lifecycle action, and should be treated
+// as the source of truth by controllers.
+type Status string
+
+const (
+	// Pending means the action has not yet executed.
+	Pending Status = "Pending"
```

Comment:
Spoke with @pmorie. When a user's using something like `osc describe` to see a deployment, they likely only care about the logical status of the hook. We decided it might make sense to introduce one more logical status which is something like `CompleteWithErrors`. So the user will see the status of the deployment, as well as the status of hooks expressed as one of the logical status values.